File: LowLevelDesign/tic-tac-toe/src/state/WinnerState.ts
import { Game } from "../Game";
import { Player } from "../Player";

export class WinnerState{
    constructor(){

    }

    handleMove(game: Game, player: Player, row: number, col: number): void {
        // No action needed as the game has already been won
        throw new Error(`Game already won by ${game.getWinner()?.name}. No more moves allowed.`);
    }
}

File: LowLevelDesign/java/tictactoe/src/com/tictac/state/WinnerState.java
package com.tictac.state;

import com.tictac.entities.Game;
import com.tictac.entities.Player;

public class WinnerState implements GameState {
    public void handleMove(Game game, Player player, int row, int col) {
        System.out.printf("Game already won by %s", game.getWinner().getName());
    }
}
